File: rust/examples/file.rs
mod utils;
use std::error::Error;
use tonic::codec::CompressionEncoding;
use utils::get_tls_options;

use aristech_tts_client::{
    get_client, synthesize,
    tts_services::{
        speech_audio_format::{Codec, Container},
        SpeechAudioFormat, SpeechRequest, SpeechRequestOption,
    },
};

#[tokio::main]
async fn main() -> Result<(), Box<dyn Error>> {
    // Load environment variables from .env file
    dotenv::dotenv()?;

    let host = std::env::var("HOST")?;
    let tls_options = get_tls_options()?;
    let mut client = get_client(host, tls_options)
        .await?
        .accept_compressed(CompressionEncoding::Gzip);
    let request = SpeechRequest {
      text: "Thanks for choosing Aristech. For more information about our products visit us at aristech.de".to_string(),
      options: Some(SpeechRequestOption {
        voice_id: std::env::var("VOICE_ID").unwrap_or("anne_en_GB".to_string()),
        audio: Some(
          SpeechAudioFormat {
            container: Container::Wav as i32,
            codec: Codec::Pcm as i32,
            ..Default::default()
          }
        ),
        ..Default::default()
      }),
      ..Default::default()
    };
    let data = synthesize(&mut client, request).await?;
    std::fs::write("output.wav", data).expect("Unable to write file");

    Ok(())
}

File: python/examples/file.py
import sys

from aristech_tts_client import TtsClient, SpeechRequest, SpeechRequestOption, SpeechAudioFormat
from utils import host, auth_token, auth_secret, root_cert, ssl, voice_id

# Use the first argument or a default text
text = sys.argv[1] if len(sys.argv) > 1 else "Thanks for choosing Aristech. For more information about our products visit us at aristech.de"

client = TtsClient(host=host, ssl=ssl, root_cert=root_cert, auth_token=auth_token, auth_secret=auth_secret)
data = client.synthesize(SpeechRequest(
    text=text,
    options=SpeechRequestOption(
        voice_id=voice_id,
        audio=SpeechAudioFormat(
            container=SpeechAudioFormat.Container.WAV,
            codec=SpeechAudioFormat.Codec.PCM
        )
    )
))
# Write the audio data to a file
with open("output.wav", "wb") as f:
    f.write(data)